app/grandchallenge/evaluation/serializers.py

```diff
@@ -74,3 +90,113 @@ class Meta:
+            "phase_pk",
+        )
+
+    def get_algorithm_model(self, obj) -> dict | None:
+        if obj.submission.algorithm_model:
+            return AlgorithmModelSerializer(
+                obj.submission.algorithm_model, context=self.context
+            ).data
+        return None
+
+    def get_algorithm_image(self, obj) -> dict:
+        return AlgorithmImageSerializer(
+            obj.submission.algorithm_image, context=self.context
+        ).data
+
+    def update(self, instance, validated_data):
+        validated_data["claimed_by"] = self.context["request"].user
+        validated_data["started_at"] = now()
+        validated_data["status"] = Evaluation.CLAIMED
+        return super().update(instance, validated_data)
+
+
+class ExternalEvaluationStatusField(ChoiceField):
+    external_to_internal_status_mapping = {
+        "Succeeded": Evaluation.SUCCESS,
+        "Failed": Evaluation.FAILURE,
+    }
+
+    def __init__(self, *args, **kwargs):
+        choices = list(self.external_to_internal_status_mapping.keys())
+        super().__init__(*args, choices=choices, **kwargs)
+
+    def to_internal_value(self, data):
+        if data in self.external_to_internal_status_mapping:
+            return self.external_to_internal_status_mapping[data]
+        raise DRFValidationError("Not a valid choice.")
+
+    def to_representation(self, value):
+        for (
+            external_value,
+            internal_value,
+        ) in self.external_to_internal_status_mapping.items():
+            if value == internal_value:
+                return external_value
+        raise DRFValidationError("Not a valid choice.")
+
+
+class ExternalEvaluationUpdateSerializer(ModelSerializer):
+    metrics = JSONField(required=False)
+    status = ExternalEvaluationStatusField()
+
+    class Meta:
+        model = Evaluation
+        fields = ("metrics", "status", "error_message")
+
+    def validate(self, data):
+        if data["status"] == Evaluation.SUCCESS and "metrics" not in data:
+            raise DRFValidationError(
+                "Metrics are required for successful evaluations."
+            )
+        if (
+            data["status"] == Evaluation.FAILURE
+            and "error_message" not in data
+        ):
+            raise DRFValidationError(
+                "An error_message is required for failed evaluations."
+            )
+        return data
+
+    def update(self, instance, validated_data):
+        if validated_data["status"] == Evaluation.SUCCESS:
+            interface = ComponentInterface.objects.get(
+                slug="metrics-json-file"
+            )
+            civ = ComponentInterfaceValue(
+                interface=interface, value=validated_data["metrics"]
+            )
+            try:
+                civ.full_clean()
+                civ.save()
+                instance.outputs.add(civ)
+            except ValidationError as e:
+                raise DRFValidationError(e)
```

Comment:
Shouldn't the CIV validation be done in `validate` and then only saved in update? This is how it should be in Django, not sure how DRF differs w.r.t. handling validation errors here.

Reply:
Hm, true, I had put this in update() because we're also updating the evaluation object there at the same time, but I see now that this makes more in validate().